cpp/open3d/io/file_format/FileXYZ.cpp

```diff
@@ -106,5 +139,40 @@ bool WritePointCloudToXYZ(const std::string &filename,
     }
 }
 
+bool WritePointCloudInMemoryToXYZ(unsigned char *&buffer,
+                                  size_t &length,
+                                  const geometry::PointCloud &pointcloud,
+                                  const WritePointCloudOption &params) {
+    try {
+        utility::CountingProgressReporter reporter(params.update_progress);
+        reporter.SetTotal(pointcloud.points_.size());
+
+        std::string content;
+        for (size_t i = 0; i < pointcloud.points_.size(); i++) {
+            const Eigen::Vector3d &point = pointcloud.points_[i];
+            std::string line = utility::FastFormatString(
+                    "%.10f %.10f %.10f\n", point(0), point(1), point(2));
+            content.append(line);
+            if (i % 1000 == 0) {
+                reporter.Update(i);
+            }
+        }
+        // nothing to report...
+        if (content.length() == 0) {
+            reporter.Finish();
+            return false;
+        }
+        length = content.length();
+        buffer = new unsigned char[length];  // we do this for the caller
+        memcpy(buffer, content.c_str(), length);
```

Comment:
Since we're using `std::free`, we can also use `std::memcpy` here for consistency.

Reply:
Fixed in e64217f254a3446818d5b51a7a8f549033b80e2f